### DISCUSSION TITLE
wire:loading <div> appears and immediately disappears

### DISCUSSION THREAD

**Discussion by timgavin** [livewire/livewire#5725] [1. Help]
I have a checkbox that updates the model using `updatedFoo()`

```html
<input wire:model="foo" type="checkbox">
```

This works, but when the checkbox is clicked the icon shows and then immediately disappears in a flash. Is this a bug or am I doing something wrong with the delay?

```html
<div class="text-sm text-green-600" wire:loading.delay.longest wire:loading.block wire:target="foo">
    <i class="fas fa-circle-check mt-3"></i> Saved
</div>
```

**Comment by joshhanley** [comment]
@timgavin if the flash is happening on page load, make sure you have Livewire styles in your app layout. Otherwise try removing the second `wire:loading.block`.

Hope this helps!

**Comment by timgavin** [comment]
@joshhanley It's not happening on load, only on click. Removing the `wire:loading.block` shows nothing after updating the model; empty white space. I've tried it with `wire:loading.class.remove` as well and it's always the same: a 50 ms flash.

**Reply by joshhanley** [reply]
@timgavin Hmm ok. Just to confirm, you have slowed your connection down in dev tools while testing it?

**Reply by timgavin** [reply]
@joshhanley When I slow it down to slow 3G the icon appears for a few seconds, then goes away. So the behavior is working, it's just super fast on a local connection and `delay.longest` isn't having any affect.

Is there something I should be looking for in the dev tools when slowing it down?

**Reply by joshhanley** [reply]
@timgavin When testing delay, I actually have a custom speed of like 0.01kbps or however slow I can get it. The other thing you can do is put a sleep in your function of a couple of seconds.

But yeah delay.longest should wait 1 second before showing and then be displayed until the request has finished.

**Reply by joshhanley** [reply]
@timgavin Sorry just saw your using wire:model so you can put the sleep in an updated hook method instead.

**Reply by timgavin** [reply]
@joshhanley Adding `sleep(1)` worked, but that's kind of hacky since `delay.longest` _should_ do it.

I tried so many other methods to get this to work earlier, including sleep, and sleep was ignored so I just assumed it wouldn't work with this. Regardless, it's working now so I'll update all of my methods with the sleep function. Thanks!

Edit: just saw your comment: where would I put sleep in the hook?

**Reply by joshhanley** [reply]
@timgavin I may be misunderstanding what your trying to achieve, but I wouldn't be adding sleep to all your functions. That's only for testing out the delay locally, then you should remove them. 

The loading will only display if the request takes longer than 1 second when using longest. So hopefully that's what your after.

**Comment by timgavin** [comment]
> The loading will only display if the request takes longer than 1 second when using longest. So hopefully that's what you're after.

@joshhanley Ok, that's my problem: I misunderstood how it works. I apologize for wasting your time on that.

What I'm trying to achieve is to show a "success message" after the model has been updated. When a user ticks the checkbox and the model has been persisted, I want to show a "saved" message.

I tried all sorts of things with Livewire and Alpine, including entangle and adding a timeout on @click, which worked but it's really hacky.

Edit: I should add that I can emit a javascript event, and maybe that's what I should do, but I was hoping there was a way without having to add custom javascript .
